### PR TITLE
Added working cell revert for invalid submissions in edit data.

### PR DESCRIPTION
**Pull request by smartguest**
This is to fix the cell edit trigger for submitting cells that only worked if it detected changes in the cell from slickgrid. Previously the cell was left alone during an invalid edit (and recognized by slickgrid as having no changes), which meant that when submitting again the trigger would not be fired, causing the invalid data in the cell to be falsely added to the grid. Now a cell revert is done and the invalid cell data is restored so that the cell is recognized as dirty again by slickgrid.

This is similar to the existing logic for invalid edits in the new row. 

Fixes https://github.com/microsoft/azuredatastudio/issues/21182

GIFS of the change in action:

Existing row:
Revert cell on existing row

New row:
Revert cell on new row

### CODE REVIEW

**Review by cheenamalhotra — APPROVED** [review]
Works as expected.

File: src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts [hunk 1]
```diff
@@ -665,6 +666,34 @@ export class EditDataGridPanel extends GridParentComponent {
 		}
 	}
 
+	private async revertCurrentCell(): Promise<void> {
+		this.cellRevertInProgress = true;
+		this.updateEnabledState(false);
+		this.dataService.revertCell(this.currentCell.row, this.currentCell.column - 1)
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts:672]: Is there a reason you're not awaiting this? There's currently not reason for this function to be async (or even return a promise) - but I don't see why you don't just await these calls and get rid of the .then's in the first place (the caller can choose to not await if needed in certain places)

Comment (cheenamalhotra) [reply src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts:672]: It maybe worth addressing all the `then's` together in a separate PR though.. there seem to be a repetitive use in this class.. @smartguest can you open a new issue to address this feedback and do full cleanup to use async/await instead?

Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts:672]: Sure, as long as this works as expected. But the code here seems to expect that the refresh will actually be completed when the promise resolves, which currently isn't true

https://github.com/microsoft/azuredatastudio/pull/23981/files#diff-e068c591ac6cf01b71fff02ee11ac75e072e6ad3770bd95223906694112fabc2R741

So if you're going to immediately go and fix all the async/awaits then that seems fine, but if that's a "backlog cleanup" item then I'd definitely suggest making sure this is going to work as expected before merging in.

Comment (cheenamalhotra) [reply src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts:672]: Might as well do all async/awaits together in single (separate) commit.

Comment (smartguest) [reply src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts:672]: I should await this, this was an oversight

**Comment by smartguest** [comment]
Converted revert functions to proper async, they've been tested to work. Other areas that have async besides these two are proper async functions as well.

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**